### PR TITLE
Front end/contract refactor

### PR DESCRIPTION
**Pull request by Grimaldus12**

### CODE REVIEW

**Comment by Grimaldus12** [comment]
I replaced the isActive attribute with an isFinished one. Basically those are equal. 
Further i fixed the calculation of the price, because it always ignored the last day (not a full day but 23h59m59s). For the moment this brings back the possibility to borrow an item for just one day but we can fix this easily. In my opinion it is not a problem to request an item for only one day but we can discuss this tomorrow.

**Comment by Grimaldus12** [comment]
For the front end: 
THe tabs 'Lent Items' and 'Borrowed items' now only display not yet finished contracts. Finished ones can be seen by clicking on a button, similar to the showRequest page for offers.

**Comment by Grimaldus12** [comment]
Everything changed was evaluated with @claussmann this morning.

**Comment by claussmann** [comment]
Eventhough we now can calculate the price for borrowing only one day, it has to be said, that we cannot borrow anything for one day only, because the offer does not except the same day for start and return date. But as @Grimaldus12 said, i think borrowing and returning at the same day is totally necessary!

File: src/main/java/de/hhu/propra/sharingplatform/model/User.java [hunk 1]
```diff
@@ -104,4 +96,14 @@ public Collection getItemsExcludeRemoved() {
         }
         return notRemovedItems;
     }
+
+    public List<Contract> getChosenContracts(boolean finished) {
```
Comment (claussmann) [inline src/main/java/de/hhu/propra/sharingplatform/model/User.java:100]: Why don't we use the repo to get finished contracts? The controller or service could add fetch all not finished contracts from the repo and add them to the model.
What you did is a better SRP solution. But is is much slower than the repo isn't it?

**Review by claussmann — APPROVED** [review]

**Pull request merged**